Add generated spells selection for casters to the game view

TheFakeCake opened the pull request:
This PR aims to add generated spell selection to the game view (issue #190).

### `<GeneratedSpells>`
The new `<GeneratedSpells>` component used in the `<GameView>` page handles the UI.

### `lores-of-magic-with-spells.json`
A new JSON data file containing all the lores of magic and their spells has been added.
- [x] Review the file and chose it's final location in the source folders

### Spell rules index
Spells have been added to the rules index.
- [x] It still lacks most of the page numbers in rule books because I did it by hand. If you could get an export from tow.whfb.app it would be much appreciated.
- [x] The [Storm Call](https://tow.whfb.app/spell/storm-call) spell has the same name as another [Storm Call](https://tow.whfb.app/special-rules/storm-call) rule, so there's a conflict in the JSON index export file. For the time being I just added a capital C to its key, but the links in the app won't be correct.

### Demo
[spell_generation_demo.webm](https://github.com/user-attachments/assets/5313b78a-c3e6-4f70-ae33-99b5b20557f5)

nthiebes commented:
Very nice! 🙌 I'm moving houses right now, so it will take a bit before I can properly review your pr.

nthiebes commented:
@TheFakeCake Hey, for some reason I thought that I already let you know that the spells are now available in the export. Apparently not! If you pull from master, the rules-index-export.json should now contain all the spells.

TheFakeCake commented:
I updated the PR to use the now available spell rules export.
I also added support for Lore of Chaos spell filtering depending on the unit's Mark of Chaos.

nthiebes commented on src/utils/unit.js:
```diff
@@ -272,3 +273,131 @@ export const getUnitOptionNotes = ({ notes, key, className, language }) => {
+  if (
+    unitHasItem(unit, "Twin Heads") ||
+    unitHasItem(unit, "Silvery Wand") ||
+    unitHasItem(unit, "Tome Of Furion")
```
Spell Familiar also adds +1 spell, and it's extremely common so it can stack.

nthiebes commented:
@TheFakeCake It works great! 🙌 I only had two comments, so I hope we're soon ready to ship it.

TheFakeCake commented:
I added support for a few more magic items and options (including Spell Familiar).

About the lores and spells translations, I can put all of them in :
- the nameMap (and use the existing ones)
- in lang files

Which would you prefer ?

TheFakeCake commented:
So, it was kind of painful to use the localization files for the translations, but it's done, though the lores of magic exist both in the i18n files and nameMap.

I tried removing them from the nameMap and use the translations instead, but the RuleLink components are so dependant on the existing system, I met a wall.

Speaking of rules link, I also had to make a new <LocalizedRuleLink> component, only used for the spells at the moment.

nthiebes commented:
I have a once in a lifetime excuse (hopefully 🤞) why it took me so long to finally merge this. The house I just moved to was hit by a lightning strike and was on fire 🔥. Not joking…

Your PR looks great, and I'll merge it now and release it next week.

TheFakeCake commented:
Woaah, sorry to read that, I hope it's ok

Don't worry about the PRs, take all the time you need
